src/contexts/Accounts/Proxies/index.tsx

```diff
@@ -0,0 +1,196 @@
+// Copyright 2023 @paritytech/polkadot-staking-dashboard authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import BigNumber from 'bignumber.js';
+import { useApi } from 'contexts/Api';
+import { useConnect } from 'contexts/Connect';
+import type { ImportedAccount } from 'contexts/Connect/types';
+import React, { useEffect, useRef, useState } from 'react';
+import type { AnyApi, MaybeAccount } from 'types';
+import { clipAddress, rmCommas, setStateWithRef } from 'Utils';
+import * as defaults from './defaults';
+import type {
+  Delegate,
+  ProxiesContextInterface,
+  Proxy,
+  ProxyAccount,
+} from './type';
+
+export const ProxiesContext = React.createContext<ProxiesContextInterface>(
+  defaults.defaultProxiesContext
+);
+
+export const useProxies = () => React.useContext(ProxiesContext);
+
+export const ProxiesProvider = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const { api, isReady, network } = useApi();
+  const { accounts, activeAccount } = useConnect();
+
+  // store the proxy accounts of each imported account.
+  const [proxies, setProxies] = useState<Array<Proxy>>([]);
+  const proxiesRef = useRef(proxies);
+
+  const [unsubs, setUnsubs] = useState<AnyApi>([]);
+  const unsubsRef = useRef(unsubs);
+
+  const [delegators, setDelegators] = useState<Array<MaybeAccount>>([]);
```

Comment:
Why is this `delegators`? We need to store `delegates`, and the above `proxies` state stores the delegators, no?

This should not be `MaybeAccount` as there will never be a `null` value being stored. In addition, we need to store more than the address here. We also need the "real" account, or delegator account alongside it.